### PR TITLE
Document impact of having different versions of a provider in the Node dependency graph

### PR DESCRIPTION
**Pull request by ringods**
## Description

Document impact of having different versions of a provider in the Node dependency graph

## Checklist:

- [x] I have reviewed the [style guide](https://github.com/pulumi/pulumi-hugo/blob/master/STYLE-GUIDE.md).
- [ ] If blogging, I have reviewed the [blogging guide](https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md).
- [ ] I have manually confirmed that all new links work.
- [ ] I added aliases (i.e., redirects) for all filename changes.
- [ ] If making css changes, I rebuilt the bundle.

### CODE REVIEW

File: themes/default/content/docs/languages-sdks/javascript/_index.md [hunk 1]
```diff
@@ -196,6 +196,91 @@ in the project root, or the environment variable `PULUMI_PREFER_YARN=true`,
 then Pulumi will use Yarn instead of NPM if the executable is available in the
 path.
 
+## Component Packages
```
Comment (scottslowe) [inline themes/default/content/docs/languages-sdks/javascript/_index.md:199]: Should this be "Package Management with Components"? Or "Managing Package Version with Components"? I'm just wondering if there's a more descriptive/informative title we could use here.

**Review by scottslowe — COMMENTED** [review]
Just one comment regarding the title; otherwise, I think this is a useful and applicable addition to the documentation.

File: themes/default/content/docs/languages-sdks/javascript/_index.md [hunk 2]
```diff
@@ -196,6 +196,91 @@ in the project root, or the environment variable `PULUMI_PREFER_YARN=true`,
...
+    ...
+    "@pulumi/random": "^4",
+    "components": "^1",
+    ...
+  },
+  ...
+}
+```
+
+In the situation where you mix major versions, the specification `^v3` excludes major `v4`. If we check on our
```
Comment (desteves) [inline themes/default/content/docs/languages-sdks/javascript/_index.md:239]: Remove the `v`. It should read `^3` and `^4` for consistency.

File: themes/default/content/docs/languages-sdks/javascript/_index.md [hunk 3]
```diff
@@ -196,6 +196,91 @@ in the project root, or the environment variable `PULUMI_PREFER_YARN=true`,
...
+When you use a Pulumi NodeJS SDK for a provider, Pulumi will always use the plugin binary of the same version.
+By having multiple versions of an SDK in our dependency graph, you will see multiple versions of the
+default provider in your Pulumi state file as a result. The child `random` resources of components from your library
+will be provisioned with plugin binary v3, while the top level `random` resources will use v4.
+
+A similar situation would happen if you would pin the component library and the consuming project to explicit
+but different versions. For instance, using `"4.11.0"` in the component library and `"4.14.0"` in the consuming
+project would result in 2 default providers with exactly these versions.
+
+If you only want to have a single version of the package and default provider, we suggest to use the carrot
```
Comment (desteves) [inline themes/default/content/docs/languages-sdks/javascript/_index.md:262]: It's "caret"

Comment (ringods) [reply themes/default/content/docs/languages-sdks/javascript/_index.md:262]: Was probably thinking too much of a snowman. 😂

File: themes/default/content/docs/languages-sdks/javascript/_index.md (hunk 1 above)
Comment (ringods) [reply themes/default/content/docs/languages-sdks/javascript/_index.md:199]: @scottslowe what about this:

> Understanding Dependencies on Provider Packages within Component Packages

Comment (scottslowe) [reply themes/default/content/docs/languages-sdks/javascript/_index.md:199]: I'm fine with that (or just "Dependencies on..." without the "Understanding")

**Review by scottslowe — APPROVED** [review]

**Pull request merged**